config/methods.config

```diff
@@ -37,22 +39,67 @@ methods {
         }
     }
 
-    determine_liftover_direction = {
-        if (params.funcotator_data.src_reference_id == params.funcotator_data.dest_reference_id) {
-            throw new IllegalArgumentException("params.funcotator_data.src_reference_id and params.funcotator_data.dest_reference_id must be different!")
-        }
+    expand_parameters = {
+        // Make sure the user didn't set any of the advanced parameters
+        def advanced_parameters = [
+            'src_fasta_id',
+            'src_fasta_ref',
+            'src_fasta_fai',
+            'src_fasta_dict',
+            'dest_fasta_id',
+            'dest_fasta_ref',
+            'dest_fasta_fai',
+            'dest_fasta_dict'
+        ]
 
-        if (![params.funcotator_data.src_reference_id, params.funcotator_data.dest_reference_id].contains("hg38")) {
-            throw new IllegalArgumentException("One of params.funcotator_data.src_reference_id and params.funcotator_data.dest_reference_id must be 'hg38'!")
+        for (key in advanced_parameters) {
+            if (params.containsKey(key)) {
+                throw new Exception("Do not directly set params.${key} - the value will be inferred from params.liftover_direction")
+            }
         }
 
-        params.liftover_forward = params.funcotator_data.dest_reference_id == "hg38"
+        def liftover_direction  = params.getOrDefault('liftover_direction', null)
+
+        def forward  = "GRCh37ToGRCh38"
+        def backward = "GRCh38ToGRCh37"
+
+        if (liftover_direction in [forward, backward]) {
+            if (liftover_direction == forward) {
```

Comment:
**question (non-blocking)**: Is the setup here with the checks and condition intended to make it extensible for potential future additions?

Reply:
Yeah, that's the idea. It's probably a little over-complicated for just the two cases, but it should be easy to update either side (the user-facing parameters or the internal parameters) without too much hassle for the other.